src/ILCompiler.Compiler/src/IL/ILImporter.Scanner.cs

```diff
@@ -111,8 +111,13 @@ public DependencyList Import()
                 if (!_isFallbackBodyCompilation &&
                     (_canonMethod.Signature.IsStatic || _canonMethod.IsConstructor || owningType.IsValueType))
                 {
-                    // For beforefieldinit, we can wait for field access.
-                    if (!((MetadataType)owningType).IsBeforeFieldInit)
+                    // For beforefieldinit, we can wait for field access...
+                    bool runCctor = !((MetadataType)owningType).IsBeforeFieldInit;
+
+                    // ...unless this is a PInvoke, in which case we run it regardless
```

Comment:
I wish I knew. This is just trying to quirk the behavior of the code RyuJIT generated for this. I don't exactly know why it triggers the cctor in this case. I don't think it should for the program above.